torch/distributions/kl.py

```diff
@@ -278,6 +279,31 @@ def _kl_laplace_laplace(p, q):
     return t1 + t2 + t3 - 1
 
 
+@register_kl(MultivariateNormal, MultivariateNormal)
+def _kl_multivariatenormal_multivariatenormal(p, q):
+    if p.batch_shape != q.batch_shape:
+        raise ValueError("KL-divergence between two Multivariate Normals with\
+                          different batch shapes cannot be computed")
+
+    def _batch_trace(bmat):
```

Comment:
nit: Could you move these outside of the `_kl_...()` function definition, so they need not create a new function each time they are called?

Reply:
Sure. Thank you.